File: 238-product-of-array-except-self/238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int res[] = new int[nums.length];
        res[0] = 1;
        int right[] = new int[nums.length];
        right[right.length - 1] = 1;
        for (int i = 1; i < res.length; i++) {
            res[i] = res[i - 1] * nums[i - 1];
        }
        for (int i = (right.length - 2); i >= 0; i--) {
            right[i] = right[i + 1] * nums[i + 1];
        }
        for (int i = 0; i < res.length; i++) {
            res[i] *= right[i];
        }
        return res;
    }
}

File: 238-product-of-array-except-self/238-product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        left = [1] * len(nums)
        right = [1] * len(nums)
        for i in range(1, len(nums), 1):
            left[i] = left[i - 1] * nums[i - 1]
        for i in range(len(nums) - 2, -1, -1):
            right[i] = right[i + 1] * nums[i + 1]
        print(left, right)
        for i in range(0, len(nums), 1):
            left[i] = left[i] * right[i]
        return left
        